pkg/sti/build.go

```diff
@@ -105,6 +104,8 @@ func (b *Builder) Build() (*api.Result, error) {
 		if err := bh.execute(api.Assemble); err != nil {
 			return nil, err
 		}
+	} else if ok && err != nil {
+		return nil, errors.NewAssembleError(bh.Request().Tag, e.Output, err)
```

Comment:
this is getting a bit hairy, I would refactor this to use switch or re-organize the if's